Do-It/Controller/CreateDoItTableViewController.swift

```diff
@@ -30,14 +31,29 @@ class CreateDoItTableViewController: UITableViewController {
 
     weak var delegate: CreateDoItTableViewControllerDelegate?
 
+    var doItToEdit: DoIt?
```

Comment:
Rather than determining adding-versus-editing using an optional DoIt, create an enum inside the CreateDoItTableViewController class:
```swift
enum InputMode {
    case addNewDoIt
    case editDoIt(DoIt)
}
```
The `editDoIt` case has an associated value containing the do-it being edited.